Library/Homebrew/extend/ARGV.rb

```diff
@@ -13,7 +13,7 @@ def flags_only
 
   def formulae
     require "formula"
-    @formulae ||= (downcased_unique_named - casks).map { |name| Formulary.factory(name, spec) }
+    @formulae ||= (downcased_unique_named - casks).map { |name| Formulary.factory_with_priority(name, spec) }
```

Comment:
Thanks for the fast response. Sorry I didn't express it clearly.

In current `Formulary#factory`, if `ref` is a formula file path then it will be used directly.

However if we change `ARGV#formulae` in the above way, the file path will be passed to `Formulary#find_with_priority` if it contains no slashes. I think the behavior would still be correct though, as there should not be formulae having name ending with `.rb`.